staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -335,6 +338,11 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 					klog.V(2).Infof("%s: watch of %v returned 429 - backing off", r.name, r.expectedTypeName)
 					<-r.initConnBackoffManager.Backoff().C()
 					continue
+				case apierrors.IsInternalError(err):
+					if retryInternalError.Retry() {
+						continue
+					}
+					fallthrough
```

Comment:
Other comment - are you sure that "no-leader" is coming only from here. Can it come from creating a watch (retries from lines 321-326)?